iocore/net/quic/QUICTypes.h

```diff
@@ -181,22 +181,15 @@ class QUICError
   }
 };
 
-class QUICNoError : public QUICError
-{
-public:
-  QUICNoError() : QUICError() {}
-};
-
 class QUICConnectionError : public QUICError
 {
 public:
   QUICConnectionError() : QUICError() {}
   QUICConnectionError(QUICTransErrorCode error_code, const char *error_msg = nullptr,
                       QUICFrameType frame_type = QUICFrameType::UNKNOWN)
     : QUICError(QUICErrorClass::TRANSPORT, static_cast<uint16_t>(error_code), error_msg), _frame_type(frame_type){};
-  QUICConnectionError(QUICErrorClass error_class, uint16_t error_code, const char *error_msg = nullptr,
-                      QUICFrameType frame_type = QUICFrameType::UNKNOWN)
-    : QUICError(error_class, error_code, error_msg), _frame_type(frame_type){};
+  QUICConnectionError(uint16_t error_code, const char *error_msg = nullptr, QUICFrameType frame_type = QUICFrameType::UNKNOWN)
+    : QUICError(QUICErrorClass::APPLICATION, error_code, error_msg), _frame_type(frame_type){};
```

Comment:
I understand what you want and I agree with this code is ugly. But unfortunately, TLS Alerts are converted to Transport Error Codes (CRYPTO_ERROR (0x1XX)). So we can't detect `QUICErrorClass` by type of `error_code`.

https://tools.ietf.org/html/draft-ietf-quic-tls-15#section-4.8
>    A TLS alert is turned into a QUIC connection error by converting the
   one-octet alert description into a QUIC error code.  The alert
   description is added to 0x100 to produce a QUIC error code from the
   range reserved for CRYPTO_ERROR.  The resulting value is sent in a
   QUIC CONNECTION_CLOSE frame.

If we go with this, we need list up all of TLS alerts.

Reply:
If we go with this change, we need list up all of TLS alerts. Maybe we can (TLS 1.3 has 34 alert message). But we need to maintain them with TLS versions which QUIC will use.

https://tools.ietf.org/html/rfc8446#appendix-B.2